tests/unit/test_job_context.py

```diff
@@ -104,3 +104,43 @@ def test_job_description_job_time(job_factory):
         job_result=job_context.JobResult(start_timestamp=20.0),
     ).job_description(current_timestamp=30.0)
     assert descr == "worker 2: some_task[12](a='b') (started 10.000 s ago)"
+
+
+def test_should_abort(app, job_factory):
+    job = job_factory(id=12)
+    app.job_manager.get_job_status = lambda job_id: jobs.Status.ABORTING
+    context = job_context.JobContext(app=app, job=job)
+
+    assert context.should_abort() is True
```

Comment:
You have implemented the InMemoryConnector. You don't need to overwrite get_job_status, you can just simulate a normal job. (same below)

Also, this and the code below only differ by `jobs.Status.ABORTING/jobs.Status.DOING` and `True/False`. Looks like it could be a single parametrized test (though this advice is probably incompatible with the one above :D You choose, but I prefer the one above.)

Reply:
Ok, I simulated the normal job cycle (with fetching and cancelling it).